hcipy/util/finite_difference.py

```diff
@@ -18,9 +20,9 @@ def generate_convolution_matrix(grid, kernel):
 	array_like
 		The matrix that applies the convolution.
 	'''
-	num_x = kernel.shape[1]
-	num_y = kernel.shape[0]
-
+	num_x = kernel.grid.shape[1]
```

Comment:
You are now relying on `kernel` being a `Field`. Change the documentation to reflect this.

Reply:
I have added a hasattr to check if it is a field.

Reply:
No need to explicitly check. It's just that the documentation still says `kernel` needs to be `array_like`, while it actually needs to be a `Field`.

Reply:
Documentation is unchanged... `kernel` can be both a Field and a ndarray, but the documentation still says only ndarray.

Reply:
Changed the documentation to "Field or array_like",